File: happyShopping_manager_web/src/main/webapp/js/service/contentService.js
//服务层
app.service('contentService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../content/findAll.action');
	};
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../content/findPage.action?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../content/findOne.action?id='+id);
	};
	//增加 
	this.add=function(entity){
		return  $http.post('../content/add.action',entity );
	};
	//修改 
	this.update=function(entity){
		return  $http.post('../content/update.action',entity );
	};
	//删除
	this.dele=function(ids){
		return $http.get('../content/delete.action?ids='+ids);
	};
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../content/search.action?page='+page+"&rows="+rows, searchEntity);
	}    	
});

File: happyShopping_content_interface/src/main/java/com/happyShopping/content/service/ContentService.java
package com.happyShopping.content.service;
import java.util.List;


import com.happyShopping.model.Content;
import entity.PageResult;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface ContentService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<Content> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(Content content);
	
	
	/**
	 * 修改
	 */
	public void update(Content content);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public Content findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(Content content, int pageNum, int pageSize);

	/**
	 * 通过广告分类的id得到广告
	 * @param categoryId
	 * @return
	 */
	public List<Content> findByCatergoryId(Long categoryId);
	
}
